backend/store/postgres/sync.go

```diff
@@ -0,0 +1,110 @@
+package postgres
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/jackc/pgx/v5"
+	"github.com/sensu/sensu-go/backend/store"
+)
+
+// SynchronizedExecutor functions as a postgresql advisory lock based mutex.
+type SynchronizedExecutor struct {
+	DB DBI
+
+	CheckinInterval time.Duration
+}
+
+// Execute blocks until the Mutex can be locked, executes the handler function,
+// then unlocks the Mutex. Implemented with postgresql advisory locks.
+func (se *SynchronizedExecutor) Execute(ctx context.Context, mux store.Mutex, handler store.MutexHandler) error {
+	var next time.Time
+	for {
+		now := time.Now()
+		if next.After(now) {
+			select {
+			case <-ctx.Done():
+				return nil
+			case <-time.After(next.Sub(now)):
+			}
+		}
+		tx, err := se.tryLock(ctx, mux)
+		if err != nil {
+			logger.WithError(err).Errorf("error trying to lock mutex: %d", mux)
+			continue
```

Comment:
It seems like in this circumstance we're going to be pretty mean to the database potentially. `next` isn't getting bumped yet, and there is no backoff on creating database transactions that could fail over and over. Perhaps the code should bump next before trying the lock, or use backoff in the retry loop.